### PR TITLE
docs(node): Node tutorial rework

### PR DESCRIPTION
**Pull request by ZackDeRose**
Reworks the node tutorial in the same way the react tutorial was reworked (see #12498)

### CODE REVIEW

File: docs/shared/node-tutorial/1-code-generation.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+---
+title: 'Node Nx Tutorial - Part 1: Code Generation'
+description: In this tutorial you'll create a backend-focused workspace with Nx.
+---
+
+{% callout type="check" title="Integrated Repo" %}
```
Comment (vsavkin) [inline docs/shared/node-tutorial/1-code-generation.md:6]: I'd reword the callout. It should be something about two styles which point to gettingstarted/intro

File: docs/shared/node-tutorial/1-code-generation.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+This tutorial sets up an [integrated](/concepts/integrated-vs-package-based) repo. If you prefer a [package-based repo](/concepts/integrated-vs-package-based), check out the [Core Tutorial](/getting-started/core-tutorial).
+{% /callout %}
+
+# Node Nx Tutorial - Part 1: Code Generation
+
+In this tutorial you'll create a backend-focused workspace with Nx.
```
Comment (vsavkin) [inline docs/shared/node-tutorial/1-code-generation.md:12]: I"d remove this sentence

File: docs/shared/node-tutorial/1-code-generation.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+- [1 - Code Generation](/node-tutorial/1-code-generation)
+- [2 - Nx Graph](/node-tutorial/2-nx-graph)
+- [3 - Task Running](/node-tutorial/3-task-running)
+- [4 - Workspace Optimization](/node-tutorial/4-workspace-optimization)
+- [5 - Summary](/node-tutorial/5-summary)
+
+## Your Objective
+
+For this tutorial, your objective is to create the initial architecture for a workspace with the following requirements:
```
Comment (vsavkin) [inline docs/shared/node-tutorial/1-code-generation.md:24]: The language seems formal so the sentence is longer it should be. 

The whole thing, including the bullet points, can be described something like that:

In this tutorial you will create a workspace with a Express.js microservice and a command line application sharing code.

File: docs/shared/node-tutorial/1-code-generation.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+- it should contain a node cli application called: `products-cli`.
+- it should contain a library for interacting with a data-source called: `products-data-client`.
+- your `products-api` app should depend on the `products-data-client` library.
+- your `products-cli` app should also depend on the `products-data-client` library.
+
+![Our Workspace Requirements](/shared/node-tutorial/requirements-diagram.png)
+
+## Creating an Nx Workspace
+
+To generate your Nx Workspace, use the [`create-nx-workspace` script from npm](https://www.npmjs.com/package/create-nx-workspace):
```
Comment (vsavkin) [inline docs/shared/node-tutorial/1-code-generation.md:36]: Reword as  "run npx create-nx-workspace@latest` :

File: docs/shared/node-tutorial/1-code-generation.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+```bash
+✔ Workspace name (e.g., org name)     · my-products
+✔ What to create in the new workspace · express
+✔ Application name                    · products-api
+```
+
+{% card title="Opting into Nx Cloud" description="You will also be prompted whether to add Nx Cloud to your workspace. We won't address this in this tutorial, but you can see the introduction to Nx Cloud for more details." url="/nx-cloud/intro/what-is-nx-cloud" /%}
+
+Once the script is complete, you can open up the `my-products` directory that was created, and see that your first project has been added to the workspace, located in `apps/products-api`.
```
Comment (vsavkin) [inline docs/shared/node-tutorial/1-code-generation.md:52]: Once the command completes

File: docs/shared/node-tutorial/1-code-generation.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+## Adding Another Application to Your Workspace
+
+Next you'll use [Nx generators](/plugin-features/use-code-generators) to generate the required `products-cli` application.
```
Comment (vsavkin) [inline docs/shared/node-tutorial/1-code-generation.md:56]: Next let's generate `product-cli`

File: docs/shared/node-tutorial/1-code-generation.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+The following syntax is used to run generators:
+
+![Nx Generator Syntax](/shared/node-tutorial/generator-syntax.png)
+
+To run the node application generator and create your `products-cli` application, run the command `npx nx g @nrwl/node:app products-cli`:
```
Comment (vsavkin) [inline docs/shared/node-tutorial/1-code-generation.md:62]: Move the command up

File: docs/shared/node-tutorial/1-code-generation.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+```bash
```
Comment (vsavkin) [inline docs/shared/node-tutorial/1-code-generation.md:64]:
```bash is not correct I think. The output isn't bash

File: docs/shared/node-tutorial/1-code-generation.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+CREATE apps/products-cli/src/main.ts
+CREATE apps/products-cli/tsconfig.app.json
+CREATE apps/products-cli/tsconfig.json
+CREATE apps/products-cli/project.json
+CREATE apps/products-cli/.eslintrc.json
+CREATE apps/products-cli/jest.config.ts
+CREATE apps/products-cli/tsconfig.spec.json
+```
+
+To see all options for the `application` generator, you can run the command `npx nx generate @nrwl/node:application --help`:
```
Comment (vsavkin) [inline docs/shared/node-tutorial/1-code-generation.md:82]: Remove this

**Review by vsavkin — CHANGES_REQUESTED** [review]
I provided some comments. Essentially, it can be shortened quite a bit.

File: docs/shared/node-tutorial/1-code-generation.md (hunk 8 above)
Comment (ZackDeRose) [reply docs/shared/node-tutorial/1-code-generation.md:64]: I'll reach out in the docs channel tomorrow AM. We had mentioned trying no language specified on the code fences, but somehow not having a language makes it looks worse:

Screen Shot 2022-10-13 at 12 44 21 AM

(`terminal` and `console` didn't work for me either)

**Comment by ZackDeRose** [comment]
Cherry-picked these commits into #12498 to consolidate to 1 PR
